pulumi/infra/postgres.py

```diff
@@ -0,0 +1,115 @@
+        name: str,
+        vpc_id: pulumi.Input[str],
+        subnet_ids: pulumi.Input[List[str]],
+        nomad_agent_security_group_id: pulumi.Input[str],
+        opts: Optional[pulumi.ResourceOptions] = None,
+    ) -> None:
+        super().__init__("grapl:Postgres", name, None, opts)
+
+        # Burstable, 2GB memory, Gravitron 2.
+        # With 5GB of storage and all-day usage, it comes out to about $24/mo
+        instance_class = "db.t4g.small"
+
+        child_opts = pulumi.ResourceOptions(parent=self)
+
+        self.username = "postgres"
+        self.password = random.RandomPassword(
+            "password",
+            length=32,
+            # Disable special characters, ":" can lead to unexpected
+            # sqlx errors: https://github.com/launchbadge/sqlx/issues/1624
+            special=False,
+            opts=child_opts,
+        ).result
+
+        rds_subnet_group = aws.rds.SubnetGroup(
+            "subnet-group",
+            subnet_ids=subnet_ids,
+            tags={"Name": f"{name}-subnet-group"},
+            opts=child_opts,
+        )
+
+        self.security_group = aws.ec2.SecurityGroup(
+            f"postgres-security-group",
+            vpc_id=vpc_id,
+            # Tags are necessary for the moment so we can look up the resource from a different pulumi stack.
+            # Once this is refactored we can remove the tags
+            tags={"Name": f"{name}-postgres-security-group"},
+            opts=pulumi.ResourceOptions(parent=self),
+        )
+        postgres_port = 5432
+
+        # Allow communication between nomad-agents and RDS
+        aws.ec2.SecurityGroupRule(
+            "nomad-agents-egress-to-rds",
+            type="egress",
+            security_group_id=nomad_agent_security_group_id,
+            from_port=postgres_port,
+            to_port=postgres_port,
+            protocol="tcp",
+            source_security_group_id=self.security_group.id,
+            opts=pulumi.ResourceOptions(parent=self.security_group),
+        )
+
+        aws.ec2.SecurityGroupRule(
+            "rds-ingress-from-nomad-agents",
+            type="ingress",
+            security_group_id=self.security_group.id,
+            from_port=postgres_port,
+            to_port=postgres_port,
+            protocol="tcp",
+            source_security_group_id=nomad_agent_security_group_id,
+            opts=pulumi.ResourceOptions(parent=self.security_group),
+        )
+
+        # There's some funkiness where, if not specified, you may end up with
+        # RDS trying to stand up an instance in an AZ not included in `subnets`.
+        def to_az(ids: List[str]) -> pulumi.Output[str]:
+            subnet_id = ids[0]
+            subnet = aws.ec2.Subnet.get("subnet", subnet_id)
+            # for some reason mypy gets hung up on the typing of this
+            az: pulumi.Output[str] = subnet.availability_zone
+            return az
+
+        availability_zone: pulumi.Output[str] = pulumi.Output.from_input(
```

Comment:
Would it be easier to just pass an availability zone into the resource constructor directly, rather than going through this?